pandas/core/groupby.py

```diff
@@ -2315,6 +2315,9 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                 # groupby
                 else:
                     cat = self.grouper.unique()
+                    cat.add_categories([c for c in self.grouper.categories
+                                        if c not in cat.categories],
```

Comment:
Kind of like this? I don't know what I'm doing. 8) flake8 diff surely doesn't pass now.